types/react/index.d.ts

```diff
@@ -857,20 +857,17 @@ declare namespace React {
     type SetStateAction<S> = S | ((prevState: S) => S);
     // this technically does accept a second argument, but it's already under a deprecation warning
     // and it's not even released so probably better to not define it.
-    type Dispatch<A> = (value: A) => void;
-    // Since action _can_ be undefined, dispatch may be called without any parameters.
-    type DispatchWithoutAction = () => void;
-    // Unlike redux, the actions _can_ be anything
     type Reducer<S, A> = (prevState: S, action: A) => S;
-    // If useReducer accepts a reducer without action, dispatch may be called without any parameters.
-    type ReducerWithoutAction<S> = (prevState: S) => S;
     // types used to try and prevent the compiler from reducing S
     // to a supertype common with the second argument to useReducer()
     type ReducerState<R extends Reducer<any, any>> = R extends Reducer<infer S, any> ? S : never;
-    type ReducerAction<R extends Reducer<any, any>> = R extends Reducer<any, infer A> ? A : never;
-    // The identity check is done with the SameValue algorithm (Object.is), which is stricter than ===
-    type ReducerStateWithoutAction<R extends ReducerWithoutAction<any>> =
-        R extends ReducerWithoutAction<infer S> ? S : never;
+    // If useReducer accepts a reducer without action, dispatch may be called without any parameters.
+    type Dispatch<A> = (value: A) => void;
```

Comment:
Yeah I'll make all of this into a single commit with only necessary changes and force push it up. I'll get this done later today.

Reply:
No need to squash in a PR. Oftentimes better to keep single commits to understand the work progression. We squash on merge anyway.

Reply:
Well I'm not super great at Git... I was reading up on how to do this and I ended up squashing just the last 3 commits and amending the remaining commit with changes back to how things were with these types/comments. I also restored the `ReducerWithoutAction` type because it actually is a useful type on its own, and that would probably be where breaking changes could occur. And now VSCode is showing the original authors on the lines I originally deleted. So I changed the commit history on my local branch, so my bad with that. I force pushed it just a minute ago. 

This is actually my first PR to an open source project. I have my own state management library (as one does) called [StateAdapt](https://state-adapt.github.io/react) but I've just been committing directly to `main` until other start contributing.

Reply:
I just realized this comment no longer totally reflects why this type exists. It's not for `useReducer` anymore. Should I update this?
```
    // If useReducer accepts a reducer without action, dispatch may be called without any parameters.
    type ReducerWithoutAction<S> = (prevState: S) => S;
```

Oh and I just remembered I've opened PRs to random small projects on GitHub. Just nothing serious that more than 100 people have starred.

Reply:
Actually that comment still applies if they use it to type a reducer defined outside of `useReducer` that's still used with `useReducer`... idk

Reply:
Yeah this was more about (needlessly) moving around types. Now we have a clear diff showing removed and added types.